utils/crypto/bls/signer/localsigner/localsigner.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package localsigner
+
+import (
+	"crypto/rand"
+	"errors"
+	"runtime"
+
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+
+	blst "github.com/supranational/blst/bindings/go"
+)
+
+var (
+	ErrFailedSecretKeyDeserialize            = errors.New("couldn't deserialize secret key")
+	_                             bls.Signer = (*LocalSigner)(nil)
+)
+
+type secretKey = blst.SecretKey
+
+type LocalSigner struct {
+	sk *secretKey
+}
+
+// NewSecretKey generates a new secret key from the local source of
+// cryptographically secure randomness.
+func New() (*LocalSigner, error) {
+	var ikm [32]byte
+	_, err := rand.Read(ikm[:])
+	if err != nil {
+		return nil, err
+	}
+	sk := blst.KeyGen(ikm[:])
+	ikm = [32]byte{} // zero out the ikm
+
+	return &LocalSigner{sk: sk}, nil
+}
+
+// ToBytes returns the big-endian format of the secret key.
+func (s *LocalSigner) ToBytes() []byte {
+	return s.sk.Serialize()
+}
+
+// SecretKeyFromBytes parses the big-endian format of the secret key into a
+// secret key.
+func SecretKeyFromBytes(skBytes []byte) (*LocalSigner, error) {
```

Comment:
Sorry to keep doing this to you
```suggestion
func FromBytes(skBytes []byte) (*LocalSigner, error) {
```